python/publish_unit_test_results.py

```diff
@@ -294,7 +294,9 @@ def get_settings(options: dict, gha: Optional[GithubAction] = None) -> Settings:
         ignore_runs=get_bool_var('IGNORE_RUNS', options, default=False, gha=gha),
         check_run_annotation=annotations,
         seconds_between_github_reads=float(seconds_between_github_reads),
-        seconds_between_github_writes=float(seconds_between_github_writes)
+        seconds_between_github_writes=float(seconds_between_github_writes),
+        job_summary=get_bool_var('JOB_SUMMARY', options, default=True, gha=gha),
+        job_summary_file=os.getenv('GITHUB_STEP_SUMMARY')
```

Comment:
There are no environment variables used anywhere in this method, only `options`, which the caller decides how to populate. In `__main__`, it is populated by `options = dict(os.environ)`, in tests it is simply a static dict.
```suggestion
        job_summary_file=options.get('GITHUB_STEP_SUMMARY')
```